src/main/kotlin/org/rust/ide/inspections/lints/RsUnnecessaryQualificationsInspection.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.lints
+
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.parentOfType
+import org.rust.ide.inspections.RsProblemsHolder
+import org.rust.ide.inspections.fixes.RemovePathPrefixFix
+import org.rust.lang.core.parser.RustParserUtil
+import org.rust.lang.core.psi.RsPath
+import org.rust.lang.core.psi.RsPathCodeFragment
+import org.rust.lang.core.psi.RsUseItem
+import org.rust.lang.core.psi.RsVisitor
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.resolve.TYPES_N_VALUES_N_MACROS
+
+class RsUnnecessaryQualificationsInspection : RsLintInspection() {
+    override fun getLint(element: PsiElement): RsLint = RsLint.UnusedQualifications
+
+    override fun buildVisitor(holder: RsProblemsHolder, isOnTheFly: Boolean): RsVisitor = object : RsVisitor() {
+        override fun visitPath(path: RsPath) {
+            val root = path.rootPath()
+            val useItem = path.parentOfType<RsUseItem>()
+
+            if (useItem == null && root == path && root.canBeShortened()) {
+                val target = getUnnecessarilyQualifiedPath(path)
+                if (target != null) {
+                    val unnecessaryPart = target.fullQualifier()
+                    val range = TextRange(0, unnecessaryPart.length)
```

Comment:
This will not work when there are spaces or comments in the middle of the path (please add such test):
```rust
pub mod a {
    pub mod b {
        pub fn func() {}
    }
}

fn main() {
    use a::b::func;
    a :: b::func();
}
```

Consider using `target.referenceNameElement!!.startOffset` here instead

Reply:
Thanks, that's much easier indeed.